problemtools/verifyproblem.py

```
@@ -1660,14 +1726,31 @@ def check(self, context: Context) -> bool:
 
         return self._check_res
 
-PROBLEM_PARTS = ['config', 'statement', 'validators', 'graders', 'data', 'submissions']
 
 class Problem(ProblemAspect):
-    def __init__(self, probdir: str):
+    """Represents a checkable problem"""
+
+    """
+    Needs a problem-format in the form of a parts-dictionary, where all classes that verify the
+    problem are listed. These should all be a subclass of ProblemPart. The dictionary is in the form
+    of category -> part-types. You could for example have 'validators' -> [InputValidators, OutputValidators].
+    """
+    def __init__(self, probdir: str, parts: dict[str, list[type]]):
+        self.part_mapping: dict[str, list[type]] = parts
+        self.aspects: set[type] = {v for s in parts.values() for v in s}
         self.probdir = os.path.realpath(probdir)
         self.shortname: str|None = os.path.basename(self.probdir)
         super().__init__(self.shortname)
         self.language_config = languages.load_language_config()
+        self.data = {}
+        self.debug(f'Problem-format: {parts}')
+
+    def get(self, part, key, default=None) -> Any:
+        if isinstance(part, type) and issubclass(part, ProblemPart):
+            part = part.PART_NAME
+        if part not in self.data:
```

This feels like a footgun to me. Do you ever use `default`, I don't think I spotted any calls with it? It feels extremely error prone that calling `get` silently returns `None` if `part not in self.data` (i.e., that part was either a typo, or wasn't initialized yet).

I would prefer if trying to access data from an uninitialized part is always an error (in problemtools, not the problem we're testing :)). If you *really* need this functionality, I think it makes sense to force the caller to explicitly toggle it on (e.g,. `get(self, part, key, default=None, crash_if_part_uninitialized=True)`)